### ISSUE TITLE
Start of the week day not correct, locale ignored

### ISSUE THREAD

**Issue by SwapFaceL** [nextcloud/calendar#2663]
### Steps to reproduce
1. Open settings and change locale from (e.g.) en_US to:
![Locale Settings](https://i.ibb.co/0nLqBSH/Unbenannt3.png)
2. Open calendar.

### Expected behaviour
Week should start with monday as indicated in the settings. Name of days should be respecting the language (they are english instead of german in this case). Popup in "Create events" should also start on monday instead of sunday.

### Actual behaviour
Week starts on sunday. Weeknames are english.

### Calendar app
2.0.4

### Client configuration
Chrome Version 84.0.4147.105 (Official Build) (64-Bit)
(same for all browsers, browsers configured to german)

### Server configuration
Ubuntu Server 20.04 LTS

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.0.4

**Nextcloud Version:** 19.0.4

**Updated from an older installed version or fresh install:** Updated

**Nextcloud configuration:**

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
 "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "19.0.4.2",
        "overwrite.cli.url":  "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "htaccess.RewriteBase": "\/",
        "maintenance": false,
        "theme": "",
        "logfile": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 2,
        "app_install_overwrite": [
            "calendar",
            "files_external_dropbox"
        ],
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "mysql.utf8mb4": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory"
    }
}

#### Browser log
![JS Log](https://i.ibb.co/vwQXTkC/Unbenannt3.png)
![JS Info Log](https://i.ibb.co/17RDK8Y/Unbenannt3.png)
![NC Settins](https://i.ibb.co/98rygMR/Unbenannt3.png)

**Comment by Redsandro** [comment]
Subscribing and +1.

Week starts on monday:
![image](https://user-images.githubusercontent.com/1702193/117139166-39c31b00-adac-11eb-9f42-5ec425b5de00.png)

Calendar starts on sunday:
![image](https://user-images.githubusercontent.com/1702193/117139257-52333580-adac-11eb-873b-e485af907173.png)

NC 21.0.1 Calendar 2.2.1.

**Comment by kominoshja** [comment]
Replicatable on v21.0.2

**Comment by MightyPork** [comment]
I found this issue searching for how to have weeks start on Monday.

Nextcloud 20.0.9, calendar 2.3.0, this setting works correctly for me (with lang/locale=German/German, as well as English/Czech, which is what I actually use). I don't see the Deutsch locale though.

Maybe 2.3.0 fixed it, or it must depend on something outside the app - locale files installed in the system or such?

**Comment by ronnicek** [comment]
Strange.. when I set lang = English, locale = English (Germany), then it´s Sunday, but if I set locale to Czech, then it´s Monday even English (Germany) says "Week starts on Monday"

**Comment by GieltjE** [comment]
Appears to be derived from the language setting, setting it to the Dutch locale still starts on sunday while setting it to the Dutch language it suddenly starts on monday....
A lot of people (myself included) really don't like their native language and will always use another while retaining the customs (e.g. weeks starting on monday).
Shouldn't this be configurable regardless of the language/locale?

**Comment by TeaDrinkingProgrammer** [comment]
I have the same problem as GieltjE, I prefer English for better language support, but prefer to keep dutch customs. As it says in the settings that the weekday starts on monday, I do expect it to honor that setting

**Comment by mischapeters** [comment]
Can this be up-voted or bumped? :)

**Comment by ChristophWurst** [comment]
Yes, add a 👍 to the original post

**Comment by adargu** [comment]
I agree to @TeaDrinkingProgrammer

The week in calendar starts on monday, but the date format is still wrong, which is also mentioned in the first post and in #2681
It is mm/dd/yyyy and shoud be dd.mm.yyyy

Language: English (US)
Locale: German (Germany)
NC: 22.2.3
Calendar: 3.0.2

![Screenshot_2021-12-17 9:58:26](https://user-images.githubusercontent.com/26898753/146518318-17bd6256-f7e4-4383-95f2-f63f542c62b4.png)

**Comment by kbmetz** [comment]
I've changed my settings to 
Language: English (British English)
Locale: English (Europe)

It says 'Week Starts on Monday' but the calendar still displays starting on Sunday. Just adding that this is also a issue for me

**Comment by uli-f** [comment]
I'd also appreciate it if the start of the week could be set independently set from the Locale.

**Comment by politas** [comment]
I'm seeing the Locale of English (Australia) showing up as "Week starts on Sunday", which is a mistake  in itself. Australia uses the ISO standard for start of week, which is Monday.

**Comment by UNICodehORN** [comment]
I see that a lot of people run into the problem. However I personally do not face the issue anymore.

Maybe it is helpful if you add more information on the reports.
I am on 22.2.7, not the most recent version but I guess in my case the issue was only in certain versions.
Os is Ubuntu on the LTS 20.04 with latest updates from official repos.

Except for updates I did not perform an special steps to fix this.

If you face the same issue please include at least your OS version and next loud server version you are using, additionally the locale might be interesting.

**Comment by uli-f** [comment]
Here is a bit more information about my configuration:
- Nextcloud 23.0.4
- Calendar 3.3.1
- Language English (US)
- Locale English (Australia) which falsely states 'Week starts on Sunday'

Even if I explicitly set the Locale to English (Europe) and it states 'Week starts on Monday' the weeks in the calendar still start with Sunday.

I am very happy to provide more information if that is helpful to track this down.

**Comment by jalbstmeijer** [comment]
Same issue:

Nextcloud 24.0.1
Calendar	3.3.2

An other observation.
I don't even have to switch locale to have the settings switch to 'Week starts on Monday'

I start with:

Language: English (United States)
Locale: English (US)

Only switching  Language to "English (British English)" is already enough to switch settings to 'Week starts on Monday'
But also in combination with Locale to "English (Europe)" the calendar stays at Sunday as start of the week.

**Comment by politas** [comment]
My calendar and user settings just changed by themselves to "Starts on Sunday", without any change of Locale. Nextcloud 22.2.8 / Calendar 3.4.0

**Comment by ant0nwax** [comment]
I agree the issue is very severe, please prioritize

on any locale with NC 24.0.2 and Calendar App just updated to 3.4.2 i am able to reproduce the issue

locale English, region English europe, Shows week starts with Monday
Calendar App still shows Sunday?
do i need to reboot the nextcloud server in order for the changes to become effective? ;)

**Comment by cbrace** [comment]
I have set my locale to "Czech" so as to temporarily solve the problem (as suggested above) until we have a proper solution. Although English (US) remains my interface language, Czech now appears in a few places, such as file age in the Files browser.

**Comment by joeysundotcom** [comment]
Can confirm there is still a problem in 25.0.3.

On some of the locales (like English(World) or English (Europe)) there is a notification below stating "Week starts on Monday", however, when opeing the calendar, the leftmost day is Sunday. It works in English (United Kingdom), Scots Gaelic or German for example.

Furthermore, the list seems to be displayed in the set language, but sorted by the name in its own language, which doesn't make sense at all.
As is: German -> English -> Spanish -> French
Cause: Deutsch -> English -> Espanol -> Francais
To make the confusion complete, The display language is shown in the respective language, but sorted by... uhm... this is a tricky one. Espanol is between English and Francais, which is correct, but Deutsch comes after that.

I mean, the latter is not a biggie. I can find my language in the list, no matter how it's sorted or displayed. Then it's set and forget... usually.

The former, however, is a bit of a doodoo. I very nearly started telling people that I messed up my appointments, before I saw that the week started on Sunday, which was a result of me setting the locale to "English (World)"

Opinionated Garbage: Whoever came up with the idea of putting one part of the weekEND at the beginning, deserves a pat on the back for their trolling ablilites.

**Comment by shalak** [comment]
Is there a workaround for this, that will allow me to use English language, but have Monday as first day of the week?

**Comment by joeysundotcom** [comment]
> Is there a workaround for this, that will allow me to use English language, but have Monday as first day of the week?

Yes. Keep the language on English, but set the locale to something else.
"English (United Kingdom)" seems to do it.
Careful though. UK is day/month/year, US is month/day/year

Loved to see ISO there... or a manual setting.

**Comment by politas** [comment]
No, there is no workaround that works for everyone. Please stop claiming that there is. Finding a locale that specifies Week starts on Monday still has the calendar starting on Sunday.

Also, the only English locale that comes up as Starts on Monday is (World), which puts the date in the annoying US MM/DD/YYYY style used by no one else in the world.

**Comment by tcitworld** [comment]
Please follow #183 for the feature to pick the first day of the week. This issue is about the locale setting not being properly followed by the calendar app.

**Comment by politas** [comment]
I was commenting on the ineffectiveness of changing locale as a workaround.

**Comment by gmejzej** [comment]
USA local here. We find date format MM/DD/YYYY annoying also! 

I wish tech companies all made an ISO 8601 date/time standard (i.e., YYYY-MM-DD Thh:mm) option in settings.

> Also, the only English locale that comes up as Starts on Monday is (World), which puts the date in the annoying US MM/DD/YYYY style used by no one else in the world.

**Comment by TeaDrinkingProgrammer** [comment]
I thought: why wait around if I can try my hand at fixing it myself 😄. I have never worked on Nextcloud though, so let's see where this goes.

**Comment by ant0nwax** [comment]
this is a plugin, thank you

Kind Regards From Phoneland

**Comment by torolito** [comment]
My calendar did not change, until I forced the locale in the config.php.
"force_locale" => "eu"

Then it worked.

**Comment by gunnarx** [comment]
Nextcloud 25.0.4
    Calendar  4.2.4
    Language English (US)
    Locale Swedish

Result:  Week starts on Monday, 24h time, ISO dates.
HOWEVER, on externally published calendar:  Week starts on Sunday, 12h time AM/PM, US date format!

Expected: At minimum, same appearance on published calendar.  But...it ought to be more flexible:

- In my view, separate the settings from locale, as many have said.
- Then, ideally, provide separate settings for the **published** calendar, so that it can be optimized for the intended viewer.

**Comment by handkerchief333** [comment]
> My calendar did not change, until I forced the locale in the config.php. "force_locale" => "eu"
> 
> Then it worked.
> 
> I'm further investigating this, because: //! locale : Basque [eu]
> 
> I'd like to have an english locale which starts with monday. But all my others where ignored so far.

This worked for me too! Thx 👏

**Comment by ant0nwax** [comment]
Thank you the config.php thingy is the workaround, it works too for me...

**Comment by ant0nwax** [comment]
actually after long time observing the config.php thing destroyed parts of the nextcloud interface showing suddenly spanish locale for example when I darag and drop files to the browser window :) so i recommend to reopen this

**Comment by foss-** [comment]
This is supposedly fixed in 4.4.0 which was scheduled for release this week. That did not happen, instead 4.3.3 was published. So probably best to wait for 4.4.0 and retest then.

**Comment by ant0nwax** [comment]
Thank you for the heads up, interesting that something is closed in 2023 even if not tested and not released... i vote for reopen then :)

**Comment by flexibeast** [comment]
Running Calendar 4.4.3 on 25.0.8, this issue is still not resolved for me.

@politas has [noted](https://github.com/nextcloud/calendar/issues/2663#issuecomment-1133791077) that "English (Australia)" should have "Week starts on Monday" instead of "... on Sunday", but setting that aside, changing to any locale that has "starts on Monday" - for example, "English (Europe)" - has no effect on the calendar, even after logging out and logging back in: the calendar still starts on Sunday.

The list of locales doesn't seem to actually include an entry for any of "English (United Kingdom)", "English (UK)", "English (Great Britain)" or "English (GB)", but if i add `force_locale => 'en_GB'` to `config.php`, the calendar then starts on a Monday.

**Comment by TeaDrinkingProgrammer** [comment]
> Thank you for the heads up, interesting that something is closed in 2023 even if not tested and not released... i vote for reopen then :)

Just to set the record straight: This issue has been tested and fixed. It might not be very clear, but the closing comment mentions #5022, which has fixed the issue for the newest version at that point, which was part of Nextcloud 26. As @foss-  says, the issue had just not been backported to 25 yet at that point.

**Comment by TeaDrinkingProgrammer** [comment]
> Running Calendar 4.4.3 on 25.0.8, this issue is still not resolved for me.
> 
> @politas has [noted](https://github.com/nextcloud/calendar/issues/2663#issuecomment-1133791077) that "English (Australia)" should have "Week starts on Monday" instead of "... on Sunday", but setting that aside, changing to any locale that has "starts on Monday" - for example, "English (Europe)" - has no effect on the calendar, even after logging out and logging back in: the calendar still starts on Sunday.
> 
> The list of locales doesn't seem to actually include an entry for any of "English (United Kingdom)", "English (UK)", "English (Great Britain)" or "English (GB)", but if i add `force_locale => 'en_GB'` to `config.php`, the calendar then starts on a Monday.

This is an interesting case. 

The issue is that the new code works as follows: It parses the locale and takes the second part. Then it takes all the setting from that **language**. For en_EU, the code takes 'EU', tries to load a language and fails, and thus defaults to English. This is because the underlying technology works with languages.

The second problem is because the settings page code is different than the code used in the Calendar. I think the settings page uses PHP to get the locale, but I am not sure.

**TLDR: For an immediate fix, you can use any locale that results in a language (E.G. en_GB).**

I agree that this is not behavior that should be allowed to happen, but I see no easy way to fix it. I think the core issue comes back to the fact that both the language and locale have both a locale and language part, which results nonsensical options: you can for instance set the language to Chinese and set the locale to English (German). This does not make any sense.
@ChristophWurst thoughts? Also, I think we should probably move this discussion to somewhere else, maybe a new issue at nextcloud/server?

**Comment by politas** [comment]
> It parses the locale and takes the second part. Then it takes all the setting from that language.

Why does it do something so obviously wrong? It's completely destroying the purpose of locales.

**Comment by TeaDrinkingProgrammer** [comment]
> > It parses the locale and takes the second part. Then it takes all the setting from that language.
> 
> Why does it do something so obviously wrong? It's completely destroying the purpose of locales.

Excuse me, I mean countries. This is why en_EU does not work, but en_DE does

**Comment by politas** [comment]
If this was ever fixed, it's broken again. If I change my locale to English(World), the only English locale that sets Week Starts On = Monday, my calendar still starts on Sunday.

The settings dialog displays the date/time formatting and the week starts on. Why is calendar making its own incorrect determination based on a failed country code look up rather than using what the user's settings specify should be used?
